homeassistant/components/bosch_shc/config_flow.py

```diff
@@ -0,0 +1,228 @@
+
+    async def async_step_reauth(self, user_input=None):
+        """Perform reauth upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(self, user_input=None):
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                data_schema=HOST_SCHEMA,
+            )
+        self.host = host = user_input[CONF_HOST]
+        self.info = await self._get_info(host)
+        return await self.async_step_credentials()
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            try:
+                self.info = info = await self._get_info(host)
+            except SHCConnectionError:
+                errors["base"] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(info["unique_id"])
+                self._abort_if_unique_id_configured({CONF_HOST: host})
+                self.host = host
+                return await self.async_step_credentials()
+
+        return self.async_show_form(
+            step_id="user", data_schema=HOST_SCHEMA, errors=errors
+        )
+
+    async def async_step_credentials(self, user_input=None):
+        """Handle the credentials step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                zeroconf = await async_get_instance(self.hass)
+                result = await self.hass.async_add_executor_job(
+                    create_credentials_and_validate,
+                    self.hass,
+                    self.host,
+                    user_input,
+                    zeroconf,
+                )
+                entry_data = {
+                    CONF_SSL_CERTIFICATE: self.hass.config.path(DOMAIN, CONF_SHC_CERT),
+                    CONF_SSL_KEY: self.hass.config.path(DOMAIN, CONF_SHC_KEY),
+                    CONF_HOST: self.host,
+                    CONF_TOKEN: result["token"],
+                    CONF_HOSTNAME: result["token"].split(":", 1)[1],
+                }
+            except SHCAuthenticationError:
+                errors["base"] = "invalid_auth"
+            except SHCConnectionError:
+                errors["base"] = "cannot_connect"
+            except SHCSessionError:
+                _LOGGER.warning("API call returned non-OK result. Wrong password?")
+                errors["base"] = "unknown"
+            except SHCRegistrationError:
+                _LOGGER.warning(
+                    "SHC not in pairing mode! Please press the Bosch Smart Home Controller button until LED starts blinking"
```

Comment:
Please break long strings around 88 characters. Maybe this message should be included in the error message above the form?